pkg/kubelet/kubelet_node_status.go

```diff
@@ -298,7 +298,12 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 		if err != nil {
 			return fmt.Errorf("failed to get node address from cloud provider: %v", err)
 		}
-		node.Status.Addresses = nodeAddresses
+		for _, address := range nodeAddresses {
+			if address.Type == api.NodeHostName {
+				return fmt.Errorf("HostnameAddress should not be set by cloudprovider.")
```

Comment:
@liggitt hmm true, it's probably a bit arbitrary. We could defer handling the hostname-override conflict and remove that restriction until the issue actually pops up (i.e. a cloud provider setting the `NodeHostName` prop and ignoring `--hostname-override`). I suspect the proper way to handle it would be to persist the hostname-override intent somewhere, but this can be dealt with in another PR.